engine/proofview.ml

```diff
@@ -120,7 +120,8 @@ type focus_context = Evar.t list * Evar.t list
    new nearly identical function everytime. Hence the generic name. *)
 (* In this version: the goals in the context, as a "zipper" (the first
    list is in reversed order). *)
-let focus_context f = f
+let focus_context (left,right) =
+  (List.map drop_state left, List.map drop_state right)
```

Comment:
@gares what does this do? You may want to add a few words in the comment above saying why we drop the state.

Reply:
Hum, not sure I was the one that wrote the code.

If this code can be called by a tactic, then I think it is wrong.
If this code is just called by top level `{` or `2:` then I think it is correct.

Put in other terms, are unfocused goals like shelved goals? (I know then are not all in the same bag, but are rather structured in order to implement the bullets semantic check).

@ppedrot @aspiwack ?

Reply:
I probably wrote it like this temporarily because I didn't know what this focusing thing was doing. Hence my question ;)

Reply:
Isn't this PR about passing a state through tactics chained with `;`? In that case, why was it needed to change the type of `focus_context` in the first place?

Reply:
That's only used by toplevel proof manipulation, not tactics, AFAICT.

Reply:
then we are good to go, thanks